### PR TITLE
Fixed typo and cpp error

### PR DESCRIPTION
**Pull request by v-fional**

### CODE REVIEW

File: AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ int WebSocketClient::sendLength(long len, char *msg)
         {
             // Actually, we only support message size less than 2^32
             int shift = i * 8 > 32 ? 32 : i * 8;
-            msg[i + 1] = (len >> shift) & 0xff;
```
Comment (ArthurMa1978) [inline AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp]: msg[i + 1] = (((unsigned long)len) >> shift) & 0xff;

Comment (v-fional) [reply AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp]: After updated the code, the issue also can repro.

Comment (ArthurMa1978) [reply AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp]: long is 32-bit, not 64-bit, so the code should be as below
         msg[0] = 127 | (1 << 7);
         msg[1] = 0;
         msg[2] = 0;
         msg[3] = 0;
         msg[4] = 0;
         msg[5] = (len >> 24) & 0xff;
         msg[6] = (len >> 16) & 0xff;
         msg[7] = (len >> 8) & 0xff;
         msg[8] = len & 0xff;
         return 9;

File: AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp [hunk 2]
```diff
@@ -350,7 +352,7 @@ WebSocketReceiveResult *WebSocketClient::receive(char *msgBuffer, int size)
             readChar(&c);
 
             int shift = (7 - i) * 8 > 32 ? 32 : (7 - i) * 8;
-            payloadLength += (c << shift);
+            payloadLength += c << (unsigned(shift));
         }
```
Comment (ArthurMa1978) [inline AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp]: readChar(&c);  // 0
readChar(&c);  // 0
readChar(&c);  // 0
readChar(&c);  // 0
readChar(&c);
payloadLength = c << 24;
readChar(&c);
payloadLength += c << 16;
readChar(&c);
payloadLength += c << 8;
readChar(&c);
payloadLength += c;

Comment (v-fional) [reply AZ3166/src/libraries/WebSocket/src/WebSocketClient.cpp]: Updated, please check it.
as of now, there is no cpp error, and it works well

**Pull request merged**